File: Adm/model/src/Locais.java
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Locais extends CRUD<Local> {

    @Override
    public void salvar() {
        try (FileWriter writer = new FileWriter("data/local.json")) {
            Gson gson = new Gson();
            gson.toJson(objetos, writer);
           
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void abrir() {
        objetos.clear();
        try (FileReader reader = new FileReader("data/local.json")) {
            Type listType = new TypeToken<List<Local>>() {}.getType();
            objetos = new Gson().fromJson(reader, listType);
            if (objetos == null) {
                objetos = new ArrayList<>(); // Inicializa a lista se o arquivo estiver vazio
            }
            
        } catch (FileNotFoundException e) {
            objetos = new ArrayList<>(); // Se o arquivo não existir, cria uma lista vazia
           
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Adm/Models/Locais.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

namespace Adm.Models
{
    public class Locais : CRUD<Local>
    {

        public override void Salvar()
        {
            if (!Directory.Exists("../../Data"))
            {
                Directory.CreateDirectory("../../Data");
            }

            try
            {
                using (StreamWriter arquivo = new StreamWriter("../../Data/local.json"))
                {
                    var dados = new List<Dictionary<string, object>>();
                    foreach (var local in objetos)
                    {
                        dados.Add(local.ToDict());  
                    }

                    string json = JsonConvert.SerializeObject(dados);
                    arquivo.Write(json);
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
        }

        public override void Abrir()
        {
            try
            {
                using (StreamReader reader = new StreamReader("../../Data/local.json"))
                {
                    string json = reader.ReadToEnd();  // Lê o conteúdo do arquivo JSON
                    var listaLocais = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(json);

                    if (listaLocais != null)
                    {
                        objetos.Clear();
                        foreach (var obj in listaLocais)
                        {
                            // Cria o objeto local a partir do dicionário
                            var local = new Local(
                                Convert.ToInt32(obj["id"]),
                                obj["local"].ToString()
                            );

                            objetos.Add(local);  // Adiciona o local à lista
                        }
                    }
                }
            }
            catch (FileNotFoundException)
            {
                // Não faz nada se o arquivo não for encontrado
            }
            catch (IOException e)
            {
        
            }
        }
    }
}
